### PR TITLE
a small number of install and unit management related fixes

### PR DESCRIPTION
**Pull request by poettering**
Basically, fixes for #5260 and #4995.

### CODE REVIEW

File: src/shared/path-lookup.c [hunk 1]
```diff
@@ -329,12 +329,18 @@ static int acquire_config_dirs(UnitFileScope scope, char **persistent, char **ru
...
                 r = user_config_dir(&a, "/systemd/user");
-                if (r < 0)
+                if (r < 0 && r != -ENXIO)
                         return r;
 
                 r = user_runtime_dir(runtime, "/systemd/user");
-                if (r < 0)
-                        return r;
+                if (r < 0) {
+                        if (r != -ENXIO)
```
Comment (keszybz) [inline src/shared/path-lookup.c:337]: The standard form
```
if (r == -ENXIO)
    ...
if (r < 0)
   ...
```
would be nicer.

**Comment by keszybz** [comment]
Doesn't seem to be working:
```
test@rawhide systemd (pr/5303)$ systemctl status sleep
● sleep.service
   Loaded: loaded (/etc/systemd/system/sleep.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2017-02-10 12:14:55 EST; 6s ago
 Main PID: 26323 (sleep)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/sleep.service
           └─26323 /bin/sleep 1000
test@rawhide systemd (pr/5303)$ sudo mv /etc/systemd/system/sleep.service{,.0}
test@rawhide systemd (pr/5303)$ systemctl status sleep
Warning: sleep.service changed on disk. Run 'systemctl daemon-reload' to reload units.
● sleep.service
   Loaded: loaded (/etc/systemd/system/sleep.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2017-02-10 12:14:55 EST; 15s ago
 Main PID: 26323 (sleep)
    Tasks: 1 (limit: 4915)
   CGroup: /system.slice/sleep.service
           └─26323 /bin/sleep 1000
test@rawhide systemd (pr/5303)$ sudo systemctl daemon-reload
test@rawhide systemd (pr/5303)$ systemctl status sleep
Failed to dump process list, ignoring: Unit sleep.service not found.
● sleep.service
   Loaded: not-found (Reason: No such file or directory)
   Active: active (running) since Fri 2017-02-10 12:14:55 EST; 27s ago
 Main PID: 26323 (sleep)
   CGroup: /system.slice/sleep.service
```

**Comment by poettering** [comment]
did you reexec pid1?

**Comment by keszybz** [comment]
Yes.

(But let me try again, it's always possible I screwed something up.)

**Comment by keszybz** [comment]
Ah, OK, I made a typo.
```
test@rawhide systemd (pr/5303)$ systemctl status sleep
● sleep.service
   Loaded: not-found (Reason: No such file or directory)
   Active: active (running) since Fri 2017-02-10 12:14:55 EST; 14min ago
 Main PID: 26323 (sleep)
   CGroup: /system.slice/sleep.service
           └─26323 /bin/sleep 1000
```
OK, so this looks good indeed.
